modules/selection/src/main/kotlin/com.r3.corda.lib.tokens.selection/config/InMemorySelectionConfig.kt

```diff
@@ -0,0 +1,60 @@
+package com.r3.corda.lib.tokens.selection.config
+
+import co.paralleluniverse.fibers.Suspendable
+import com.r3.corda.lib.tokens.selection.selectors.LocalTokenSelector
+import com.r3.corda.lib.tokens.selection.selectors.Selector
+import com.r3.corda.lib.tokens.selection.services.VaultWatcherService
+import net.corda.core.cordapp.CordappConfig
+import net.corda.core.cordapp.CordappConfigException
+import net.corda.core.node.ServiceHub
+import org.slf4j.LoggerFactory
+
+const val CACHE_SIZE_DEFAULT = 1024 // TODO Return good default, for now it's not wired, it will be done in separate PR.
+
+// TODO that interface should be separate, same for selector
+interface StateSelectionConfig {
+    @Suspendable
+    fun toSelector(services: ServiceHub): Selector
+}
+
+data class InMemorySelectionConfig(val indexingStrategy: VaultWatcherService.IndexingType, val cacheSize: Int = CACHE_SIZE_DEFAULT) : StateSelectionConfig {
+    companion object {
+        val logger = LoggerFactory.getLogger("inMemoryConfigSelectionLogger")
+        fun parse(config: CordappConfig): InMemorySelectionConfig {
+            val cacheSize = config.getIntOrNull("stateSelection.inMemory.cacheSize")
+                    ?: CACHE_SIZE_DEFAULT
+            val indexingType = try {
+                VaultWatcherService.IndexingType.valueOf(config.get("stateSelection.inMemory.indexingStrategy").toString().toUpperCase())
+            } catch (e: CordappConfigException) {
+                VaultWatcherService.IndexingType.PUBLIC_KEY // Default is public key.
+            }
+            logger.info("Found in memory token selection configuration with values indexing strategy: $indexingType, cacheSize: $cacheSize")
+            return InMemorySelectionConfig(indexingType, cacheSize)
+        }
+    }
+
+    @Suspendable
+    override fun toSelector(services: ServiceHub): LocalTokenSelector {
+        return try {
+            val vaultObserver = services.cordaService(VaultWatcherService::class.java)
+            LocalTokenSelector(services, vaultObserver, state = null)// TODO allowShortFall and autoUnlockDelay, should it be config or per flow?
```

Comment:
probably per flow but set to a sensible default?

Reply:
I would argue that we don't need allowShortfall, I don't see sensible use case for it.
Also we have sensible default already.